src/d/actor/d_a_ff.cpp

```diff
@@ -1,55 +1,361 @@
-//
-// Generated by dtk
-// Translation Unit: d_a_ff.cpp
-//
+/**
+ * d_a_ff.cpp
+ * Object - Firefly (not collectable)
+ */
 
 #include "d/actor/d_a_ff.h"
+#include "d/actor/d_a_player.h"
+#include "d/d_bg_s_gnd_chk.h"
+#include "d/d_com_inf_game.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
+#include "d/d_s_play.h"
+#include "m_Do/m_Do_lib.h"
+#include "d/res/res_ff.h"
+
+static s32 ff_count = 0;
 
 /* 00000078-000001E4       .text fire_fly_draw__FP8ff_class */
-void fire_fly_draw(ff_class*) {
-    /* Nonmatching */
+static void fire_fly_draw(ff_class* i_this) {
+    MtxTrans(i_this->current.pos.x, i_this->current.pos.y, i_this->current.pos.z, false);
+    MtxScale(i_this->mScale, i_this->mScale, i_this->mScale, true);
+    i_this->mpModel[0]->setBaseTRMtx(*calc_mtx);
+    if (i_this->mScale > 0.01f) {
+        i_this->mBrkAnm[0]->play();
+        i_this->mBrkAnm[0]->entry(i_this->mpModel[0]->getModelData());
+        g_dComIfG_gameInfo.drawlist.setOpaListMaskOff();
+        g_dComIfG_gameInfo.drawlist.setXluListMaskOff();
+        mDoExt_modelUpdate(i_this->mpModel[0]);
+        if (i_this->mGlowScale > 0.01f) {
+            mDoMtx_YrotM(*calc_mtx, i_this->mLiveTimer << 8);
```

Comment:
If it still matches this makes more sense as multiplication
```suggestion
            mDoMtx_YrotM(*calc_mtx, i_this->mLiveTimer * 0x100);
```